drivers/ethernet/nxp_enet/Kconfig

```diff
@@ -1,34 +1,91 @@
-# ETH_MCUX Ethernet driver configuration options
+# NXP ENET ethernet drivers configuration options
 
 # Copyright (c) 2016-2017 ARM Ltd
+# Copyright 2024 NXP
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig ETH_MCUX
+menu "NXP ENET"
+
+choice NXP_ENET_DRIVER
+	prompt "NXP ENET Driver Version"
+	default ETH_NXP_ENET if DT_HAS_NXP_ENET_MAC_ENABLED
+	default ETH_MCUX if DT_HAS_NXP_KINETIS_ETHERNET_ENABLED
+	depends on DT_HAS_NXP_ENET_MAC_ENABLED || DT_HAS_NXP_KINETIS_ETHERNET_ENABLED
+
+config ETH_NXP_ENET
+	bool "NXP ENET Ethernet driver"
+	select NOCACHE_MEMORY if HAS_MCUX_CACHE
+	select ARM_MPU if CPU_CORTEX_M7
+	select MDIO if DT_HAS_NXP_ENET_MDIO_ENABLED
+	select NET_POWER_MANAGEMENT if (PM_DEVICE && SOC_FAMILY_KINETIS)
+	help
+	  Enable NXP ENET Ethernet driver.
+
+config ETH_MCUX
 	bool "MCUX Ethernet driver"
-	default y
-	depends on DT_HAS_NXP_KINETIS_ETHERNET_ENABLED
 	select NOCACHE_MEMORY if HAS_MCUX_CACHE && CPU_HAS_DCACHE
 	select ARM_MPU if CPU_CORTEX_M7
 	select NET_POWER_MANAGEMENT if PM_DEVICE
+	select DEPRECATED
```

Comment:
We should add something to the release notes highlighting this change

Reply:
agreed. Need to add to the release notes and probably the migration guide (maybe)

Reply:
Yes I was thinking about this I just haven't got to it yet (PR has a few loose ends I need to address in the morning)